src/schemathesis/runner/impl/core.py

```diff
@@ -156,9 +158,12 @@ def network_test(
     request_timeout: Optional[int],
     store_interactions: bool,
     headers: Optional[Dict[str, Any]],
+    case_hook: Optional[Callable[[Case], Case]],
 ) -> None:
     """A single test body that will be executed against the target."""
     # pylint: disable=too-many-arguments
+    if case_hook:
+        case = case_hook(deepcopy(case))
```

Comment:
There is no need for `deepcopy` since it is a unique object that is created for this test